test/test_network.py

```diff
@@ -48,33 +51,42 @@ def test_send(self):
         self.assertEqual(msg.arbitration_id, 0x12345)
         self.assertTrue(msg.is_extended_id)
 
-        bus.shutdown()
-        self.network.disconnect()
+    def test_send_periodic(self):
+        from threading import Event
```

Comment:
This changes test methodology in two ways AFAICT: Not checking the amount of packets received, and using `network.subscribe()` instead of `bus.recv()`.  Is there a good reason for the latter?  In #504 you do use a separate `ThreadSafeBus` to receive the messages from the virtual interface.  Shouldn't the same apply here?

As for the import, I see no necessity to do that locally instead of at the top of the module.

Reply:
I agree we should still check the number of messages received. Let me work a little bit more on this; marking it as draft for now.

Reply:
I think 0af94f7ed1b35266ac9a6b1344467f96f09ab4c6 should work pretty well. It checks the number of packets received, and roughly checks the period. Should be pretty deterministic.

Reply:
How certain are we that it won't catch two periodic messages?  The new check depends on exactly two packages being received.

Reply:
And what about the first comment, using `network.subscribe()` instead of `bus.recv()`?

Reply:
> How certain are we that it won't catch two periodic messages? The new check depends on exactly two packages being received.

Pretty certain, but I guess with a severely overloaded runner, you could possibly end up in a place where the background periodic task gets to shoot twice before we get to the check.

> And what about the first comment, using `network.subscribe()` instead of `bus.recv()`?

Sure, we can use `recv()` instead, I just thought `subscribe()` might be easier to read. It does not really matter which tool we use, as long as we're testing the right thing. I can draft up a proposal using `recv` instead.

Reply:
> Pretty certain, but I guess with a severely overloaded runner, you could possibly end up in a place where the background periodic task gets to shoot twice before we get to the check.

Then let's just wait and see how often that actually happens.

> I can draft up a proposal using `recv` instead.

My concern was mainly that we now implicitly test `.subscribe()` as well, which we didn't before.

Reply:
Got it.

I've got the current PR running in an infinite loop in the background; no flakiness observed (yet).

All of the experiments I've done with `.recv()` have been extremely flaky, so if you're fine with using `.subscribe()`, I'd prefer that (at least for now). It's a little bit verbose, but it's deterministic.

Reply:
Sure, there's nothing wrong with it.  Just wanted to raise awareness of using the additional method in the test.  I'm not strict about that though, any failures will have to be investigated by looking at the whole stack anyway.